campfire/src/web/build.rs

```diff
@@ -19,31 +14,75 @@ pub(crate) enum Target {
 pub struct BuildOptions {
     #[arg(long, default_value = "dev")]
     pub profile: String,
+    #[arg(long, default_value = "pkg")]
+    pub pkg_name: String,
     #[arg(long, value_enum, default_value = "bundler")]
     target: Target,
 }
 
-pub async fn run(opts: BuildOptions) -> anyhow::Result<()> {
-    ensure_wasm_pack().await?;
-    let output_path = run_cargo_build(&opts).await?;
+pub fn run(opts: BuildOptions) -> anyhow::Result<()> {
+    ensure_wasm_pack()?;
+
+    let output_path = run_cargo_build(&opts)?;
 
     eprintln!("Built package: {:?}", output_path);
 
     Ok(())
 }
 
-pub async fn ensure_wasm_pack() -> anyhow::Result<()> {
+#[cfg(not(target_os = "linux"))]
+pub(crate) fn install_wasm_pack() -> anyhow::Result<()> {
+    eprintln!("Installing wasm-pack from source");
+    let status = Command::new("cargo")
+        .args(["install", "wasm-pack"])
+        .spawn()?
+        .wait()?;
+
+    if !status.success() {
+        anyhow::bail!("Failed to install wasm-pack");
+    }
+
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+pub(crate) fn install_wasm_pack() -> anyhow::Result<()> {
+    eprintln!("Installing wasm-pack");
+    let mut curl = Command::new("curl")
+        .args([
+            "https://rustwasm.github.io/wasm-pack/installer/init.sh",
+            "-sSf",
+        ])
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .context("Failed to spawn curl")?;
+
+    let mut sh = Command::new("sh")
+        .stdin(std::process::Stdio::from(curl.stdout.take().unwrap()))
+        .spawn()
+        .context("Failed to spawn sh")?;
+
+    let sh = sh.wait()?;
+
+    let curl = curl.wait()?;
+
+    if !curl.success() {
+        anyhow::bail!("Failed to fetch install script")
+    }
+
+    if !sh.success() {
+        anyhow::bail!("Failed to run install script for wasm-pack")
+    }
+
+    Ok(())
+}
+
+pub fn ensure_wasm_pack() -> anyhow::Result<()> {
     match which::which("wasm-pack") {
         Err(_) => {
-            eprintln!("Installing wasm-pack");
-            let status = Command::new("carg")
```

Comment:
Was there a never spotted typo here?

Reply:
Good catch. It wasn't spotted as that is the macos install